### DISCUSSION TITLE
Still showing localhost:1313 un menu links then sample site is published on my site

### DISCUSSION THREAD

**Discussion by VitalijsVickops** [thuliteio/doks#657] [Q&A]
I've added default sample of hugo+doks into docker and put it on Azure. First link https://<my domain>/docs/prologue/introduction/ opens properly, but all menu items : Prologue/Introduction/Quick Start/Commands have https://localhost:1313/ URL. What I did wrong?

**Comment by h-enk** [comment]
Are you serving the results of the `npm run build` command? And, did you make sure to update `baseURL` in `./config/_default/config.toml` to your production URL?

**Reply by VitalijsVickops** [reply]
Yes and Yes. Actually, phisically all pages are ok, it is just like a menu (and actually sitemap.xml) links contain localhost:1313

**Reply by h-enk** [reply]
@ImaCrea, you're using Doks w/ Docker — could you be of any help?

**Reply by h-enk** [reply]
Could you try setting `baseurl` in `./config/production/config.toml`?

**Reply by h-enk** [reply]
Yes, but is it also explicitly set in `./config/production/config.toml` (next to `./config/_default/config.toml`) — just to rule things out?

**Reply by h-enk** [reply]
Could also be a caching thing — [docker builder prune](https://docs.docker.com/engine/reference/commandline/builder_prune/)

**Reply by VitalijsVickops** [reply]
> Yes, but is it also explicitly set in `./config/production/config.toml` (next to `./config/_default/config.toml`) — just to rule things out?

Haven't noticed it is about /production/ now, will try.

**Reply by VitalijsVickops** [reply]
Set in /production/ as well, still links and sitemap.xml values contain localhost and still pages itself are ok. It is like baseUrl value should be used by hugo to set lunks and sitemap.xml?

**Reply by h-enk** [reply]
> It is like baseUrl value should be used by hugo to set lunks and sitemap.xml?

Yes, that's correct. So, the behavior your seeing is peculiar. if `sitemap.xml` contains `localhost:1313`, the default (Hugo) baseURL is used (and not the one set in the config files). Not sure what's going on.

You could also try adding `--baseURL=<https://your.prod.url>` (takes precedence over the config one) to the `npm run build` script, or run manually:

```bash
npx exec-bin node_modules/.bin/hugo/hugo --gc --minify --baseURL=<https://your.prod.url>
```

Did you clear the cache (guess so)?

**Reply by VitalijsVickops** [reply]
Sorry if I am missing something, just to start from the very beginning, no dockerization, just local actions: 
1) npx exec-bin node_modules/.bin/hugo/hugo --gc --minify --baseURL=https://www.google.com
2) hugo server
I was expecting to see links like https://www.google.com/docs/prologue/quick-start/ (not expected to open, of course :-)) on Docs page and in sitemap.xml, but still can see only: 
http://localhost:1313/docs/prologue/quick-start/
Of course - incognito mode.
So, if these basics are not woking, obviously I misunderstood something. Any advice?

**Reply by h-enk** [reply]
Cannot reproduce your issue.

Fresh Doks install, setting `baseurl = "https://google.com/"` in `config\_default\config.toml`, run `npm run build`, serving locally (using [http-server](https://github.com/http-party/http-server)):

![Snag_14a192a4](https://user-images.githubusercontent.com/3902872/152939067-8612ace6-7734-4a1f-85fc-ad2779845df1.png)

Could you try w/ a fresh install?

**Reply by h-enk** [reply]
Don't use [Hugo server](https://gohugo.io/commands/hugo_server/) (limited options), but [http-server](https://github.com/http-party/http-server) or [Caddy](https://github.com/caddyserver/caddy)

**Reply by VitalijsVickops** [reply]
This is what I did: 
git clone https://github.com/h-enk/doks-child-theme.git doks-original  
cd doks-original  
npm install    
npx exec-bin node_modules/.bin/hugo/hugo --gc --minify --baseURL=https://your.prod.url
npm run start 

Still localhost in all links, for instance:
```
<url>
<loc>http://localhost:1313/blog/say-hello-to-doks/</loc>
<lastmod>2020-11-04T09:19:42+01:00</lastmod>
<changefreq>weekly</changefreq>
<priority>0.5</priority>
</url>
```

**Reply by VitalijsVickops** [reply]
Just to be sure tried both themes: git clone https://github.com/h-enk/doks.git doks-original  - result is the same.

**Reply by h-enk** [reply]
> npm run start
>
> Still localhost in all links, for instance:

That's default Hugo behavior /to be expected.

Like I said:

> Don't use [Hugo server](https://gohugo.io/commands/hugo_server/) (limited options), but [http-server](https://github.com/http-party/http-server) or [Caddy](https://github.com/caddyserver/caddy)

So don't use `npm run start`, but `npm run build` and next `http-server` (for instance) — if you want to preview the production site

**Reply by VitalijsVickops** [reply]
Thanks, it works now. Is it like hugo (by using npm run build) still prepares everything, but then http-server just serves it? 
One more (and the last I hope) thing - it is fine with sitemap.xml and side menu (like Prologue / [Introduction] / [Quick Start] / [Commands] etc), but header menu still leads me to http://127.0.0.1:8080/... links.

**Reply by h-enk** [reply]
>  Is it like hugo (by using npm run build) still prepares everything, but then http-server just serves it?

Yes, but it's a different prep for production (think minifying, purging CSS, setting the production URL 😉 et cetera)

> but header menu still leads me to http://127.0.0.1:8080/... links.

Yes, It takes some more to mimic the production website fully. I recommend not to go there. Your process s/b: develop locally + use Netlify's (~~if using Netlify~~ edit: you're using Docker + Azure, not sure how to do it in such a setup: staging environment?) Deploy Preview and Branch Deploy to verify in production (before promoting it to production)

**Reply by h-enk** [reply]
Just curious, any particular reason you're running Doks using Docker + Azure (is Netlify /Vercel et cetera not an option for you)?

**Reply by VitalijsVickops** [reply]
> Just curious, any particular reason you're running Doks using Docker + Azure (is Netlify /Vercel et cetera not an option for you)?

just would like to keep Doks based API description next to API itself :)

**Reply by VitalijsVickops** [reply]
Just in case, successfully put site on docker, as it was advised, "compiled" with Hugo and deployed http-server

**Reply by h-enk** [reply]
Ah, good to hear!
